test/story/story.LP.test.ts

```diff
@@ -0,0 +1,151 @@
+import BN from "bn.js";
+import { Keyring } from "@polkadot/api";
+import { AssetWallet, User } from "../../utils/User";
+import { Assets } from "../../utils/Assets";
+import { getEnvironmentRequiredVars } from "../../utils/utils";
+import { getEventResultFromMangataTx } from "../../utils/txHandler";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import { Mangata } from "mangata-sdk";
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(1500000);
+
+const defaultCurrecyValue = new BN(250000);
+
+describe("Story tests > LP", () => {
+  let testUser1: User;
+  let testUser2: User;
+  let sudo: User;
+
+  let keyring: Keyring;
+  let token1: BN;
+
+  beforeAll(async () => {
+    try {
+      getApi();
+    } catch (e) {
+      await initApi();
+    }
+  });
+
+  beforeEach(async () => {
+    keyring = new Keyring({ type: "sr25519" });
+
+    // setup users
+    testUser1 = new User(keyring);
+    testUser2 = new User(keyring);
+
+    sudo = new User(keyring, sudoUserName);
+    // add users to pair.
+    keyring.addPair(testUser1.keyRingPair);
+    keyring.addPair(testUser2.keyRingPair);
+    keyring.addPair(sudo.keyRingPair);
+    [token1] = await Assets.setupUserWithCurrencies(
+      testUser1,
+      [defaultCurrecyValue],
+      sudo
+    );
+    await testUser1.addMGATokens(sudo);
+    await sudo.mint(MGA_ASSET_ID, testUser2, defaultCurrecyValue);
+    await sudo.mint(token1, testUser2, defaultCurrecyValue);
+    testUser1.addAsset(MGA_ASSET_ID);
+    testUser2.addAsset(MGA_ASSET_ID);
+    testUser2.addAsset(token1);
+    testUser1.refreshAmounts();
+    testUser2.refreshAmounts();
+  });
+
+  test("Pool wins over 5% tokens when 10 swaps are done in the pool [Token - MGA]", async () => {
+    //lets create a pool with user1
+    const mangata = await getMangataInstance();
+    await mangata.createPool(
+      testUser1.keyRingPair,
+      token1.toString(),
+      defaultCurrecyValue,
+      MGA_ASSET_ID.toString(),
+      defaultCurrecyValue
+    );
+
+    const poolBalanceBeforeSwaps = await getBalanceOfPool(token1, MGA_ASSET_ID);
+    //lets swap tokens
+    await do10Swaps(mangata, testUser2, token1);
+    const poolBalance = await getBalanceOfPool(token1, MGA_ASSET_ID);
+    await testUser1.refreshAmounts(AssetWallet.AFTER);
+    await testUser2.refreshAmounts(AssetWallet.AFTER);
+
+    const userTokensBefore = testUser2
+      .getAsset(token1)
+      ?.amountBefore.free!.add(
+        testUser2.getAsset(MGA_ASSET_ID)?.amountBefore.free!
+      );
+
+    const userTokensAfter = testUser2
+      .getAsset(token1)
+      ?.amountAfter.free!.add(
+        testUser2.getAsset(MGA_ASSET_ID)?.amountAfter.free!
+      );
+    const totalPoolBefore = poolBalanceBeforeSwaps[0].add(
+      poolBalanceBeforeSwaps[1]
+    );
+    const totalPoolAfter = poolBalance[0].add(poolBalance[1]);
```

Comment:
Feddback from Stano: this does not make sense.

Reply:
Let's do the maths with one token:
250,000MGA - 250,000 USD
1MGA-1USD
do 10 swaps ->
TODO:Do reverse 10 swaps to get a balance pool
do 10 swaps <-
xMGA - yUSD
350,000-178,748.7522
lefSide x times right side.

Reply:
That worked. 
the 5% was an appreciation token issue.